examples/charRNN/model.go

```diff
@@ -262,7 +262,7 @@ func (r *charRNN) fwd(srcIndex int, prev *lstmOut) (retVal *lstmOut, err error)
 		prevCells = prev.cells
 	}
 
-	inputVector := r.inputVector
+	var inputVector *Node
```

Comment:
+1

Reply:
I'm sorry, I don't get why this fails if `len(r.ls) == 0`. `inputVector` is only used in the loop, and the loop won't be executed if `r.ls`'s length is 0.

However, if `len(r.ls) == 0` `hiddens` and `cells` will be empty slices, and the assignment after the loop `lastHidden := hiddens[len(hiddens)-1]` will fail since `len(hiddens)` will be 0.

I don't really know what the function should do if `len(r.ls) == 0`. What is the correct handling? Returning an error early, maybe?

Reply:
Oh, and a side note, IMHO using named return values is confusing and adding little value in this case.

Reply:
Oh, apologies, you are right, my suggestion is useless